### PR TITLE
JDK-8301068: Replace NULL with nullptr in share/jvmci/

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory share/jvmci/. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301068](https://bugs.openjdk.org/browse/JDK-8301068): Replace NULL with nullptr in share/jvmci/

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Tom Rodriguez](https://openjdk.org/census#never) (@tkrodriguez - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12183/head:pull/12183` \
`$ git checkout pull/12183`

Update a local copy of the PR: \
`$ git checkout pull/12183` \
`$ git pull https://git.openjdk.org/jdk pull/12183/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12183`

View PR using the GUI difftool: \
`$ git pr show -t 12183`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12183.diff">https://git.openjdk.org/jdk/pull/12183.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/jvmci/jvmciCompilerToVM.cpp [hunk 1]
```diff
@@ -1292,14 +1292,14 @@ void resync_vframestream_to_compiled_frame(vframeStream& vfst, intptr_t* stack_p
 }
 
 /*
- * Used by c2v_iterateFrames. Returns an array of any unallocated scope objects or NULL if none.
+ * Used by c2v_iterateFrames. Returns an array of any unallocated scope objects or nullptr if none.
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciCompilerToVM.cpp:1295]: Fix

File: src/hotspot/share/jvmci/jvmciCompilerToVMInit.cpp [hunk 2]
```diff
@@ -152,7 +152,7 @@ void CompilerToVM::Data::initialize(JVMCI_TRAPS) {
   vm_page_size = os::vm_page_size();
 
 #define SET_TRIGFUNC(name)                                      \
-  if (StubRoutines::name() != NULL) {                           \
+  if (StubRoutines::name() != nullptr) {                           \
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciCompilerToVMInit.cpp:155]: Align

File: src/hotspot/share/jvmci/jvmciEnv.cpp [hunk 3]
```diff
@@ -1689,7 +1689,7 @@ nmethod* JVMCIEnv::get_nmethod(JVMCIObject obj) {
       return (JVMCIObjectArray) wrap(array);                                                                         \
     } else {                                                                                                         \
       JNIAccessMark jni(this);                                                                                       \
-      jobjectArray result = jni()->NewObjectArray(length, JNIJVMCI::className::clazz(), NULL);                       \
+      jobjectArray result = jni()->NewObjectArray(length, JNIJVMCI::className::clazz(), nullptr);                       \
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciEnv.cpp:1692]: Align

File: src/hotspot/share/jvmci/jvmciEnv.hpp [hunk 4]
```diff
@@ -47,7 +47,7 @@ class JVMCIRuntime;
     if (env->ExceptionCheck()) { \
       if (env != JavaThread::current()->jni_environment()) { \
         char* sl_path; \
-        if (::JVMCI::get_shared_library(sl_path, false) != NULL) { \
+        if (::JVMCI::get_shared_library(sl_path, false) != nullptr) { \
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciEnv.hpp:50]: Align

File: src/hotspot/share/jvmci/jvmciEnv.hpp [hunk 5]
```diff
@@ -64,11 +64,11 @@ class JVMCIKlassHandle : public StackObj {
   Thread*    _thread;
 
   Klass*        klass() const                     { return _klass; }
-  Klass*        non_null_klass() const            { assert(_klass != NULL, "resolving NULL _klass"); return _klass; }
+  Klass*        non_null_klass() const            { assert(_klass != nullptr, "resolving nullptr _klass"); return _klass; }
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciEnv.hpp:67]: null

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp [hunk 6]
```diff
@@ -486,11 +486,11 @@ void JNIJVMCI::initialize_ids(JNIEnv* env) {
 
 #define DO_BOX_CLASS(klass, basicType, type) \
   current_class_name = "java/lang/" #klass;                                                                       \
-  if (JVMCILibDumpJNIConfig == NULL) {                                                                            \
+  if (JVMCILibDumpJNIConfig == nullptr) {                                                                            \
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:489]: Align

File: src/hotspot/share/jvmci/jvmciJavaClasses.hpp [hunk 7]
```diff
@@ -148,7 +148,7 @@
   end_class                                                                                                   \
   start_class(JavaConstant, jdk_vm_ci_meta_JavaConstant)                                                      \
     static_object_field(JavaConstant, ILLEGAL, "Ljdk/vm/ci/meta/PrimitiveConstant;")                          \
-    static_object_field(JavaConstant, NULL_POINTER, "Ljdk/vm/ci/meta/JavaConstant;")                          \
+    static_object_field(JavaConstant, nullptr_POINTER, "Ljdk/vm/ci/meta/JavaConstant;")                          \
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciJavaClasses.hpp:151]: Align

File: src/hotspot/share/jvmci/jvmciJavaClasses.hpp [hunk 8]
```diff
@@ -291,7 +291,7 @@ class JVMCIEnv;
   static void check(oop obj, const char* field_name, int offset);                    \
   static InstanceKlass* _klass;                                                      \
  public:                                                                             \
-  static InstanceKlass* klass() { assert(_klass != NULL, "uninit"); return _klass; } \
+  static InstanceKlass* klass() { assert(_klass != nullptr, "uninit"); return _klass; } \
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciJavaClasses.hpp:294]: Align

File: src/hotspot/share/jvmci/jvmciRuntime.cpp [hunk 9]
```diff
@@ -271,7 +271,7 @@ JRT_ENTRY_NO_ASYNC(static address, exception_handler_for_pc_helper(JavaThread* c
     assert(exception_frame.is_deoptimized_frame(), "must be deopted");
     pc = exception_frame.pc();
   }
-  assert(exception.not_null(), "NULL exceptions should be handled by throw_exception");
+  assert(exception.not_null(), "nullptr exceptions should be handled by throw_exception");
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciRuntime.cpp:274]: str

File: src/hotspot/share/jvmci/jvmciRuntime.cpp [hunk 10]
```diff
@@ -333,8 +333,8 @@ JRT_ENTRY_NO_ASYNC(static address, exception_handler_for_pc_helper(JavaThread* c
 
   // If the stack guard pages are enabled, check whether there is a handler in
   // the current method.  Otherwise (guard pages disabled), force an unwind and
-  // skip the exception cache update (i.e., just leave continuation==NULL).
-  address continuation = NULL;
+  // skip the exception cache update (i.e., just leave continuation==null).
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciRuntime.cpp:336]: Comparison use nullptr

File: src/hotspot/share/jvmci/jvmciRuntime.cpp [hunk 11]
```diff
@@ -583,8 +583,8 @@ JRT_END
 JRT_LEAF(void, JVMCIRuntime::log_object(JavaThread* thread, oopDesc* obj, bool as_string, bool newline))
   ttyLocker ttyl;
 
-  if (obj == NULL) {
-    tty->print("NULL");
+  if (obj == nullptr) {
+    tty->print("nullptr");
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciRuntime.cpp:587]: null

File: src/hotspot/share/jvmci/jvmci_globals.hpp [hunk 12]
```diff
@@ -129,10 +129,10 @@ class fileStream;
           "Maximum size of a compiled method.")                             \
           range(0, max_jint)                                                \
                                                                             \
-  product(ccstr, JVMCILibPath, NULL, EXPERIMENTAL,                          \
+  product(ccstr, JVMCILibPath, nullptr, EXPERIMENTAL,                          \
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmci_globals.hpp:132]: Align

File: src/hotspot/share/jvmci/vmStructs_jvmci.hpp [hunk 13]
```diff
@@ -30,24 +30,24 @@
 class JVMCIVMStructs {
 public:
   /**
-   * The last entry has a NULL fieldName.
+   * The last entry has a nullptr fieldName.
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/vmStructs_jvmci.hpp:33]: null

File: src/hotspot/share/jvmci/vmStructs_jvmci.hpp [hunk 14]
```diff
@@ -30,24 +30,24 @@
...
    */
   static VMStructEntry localHotSpotVMStructs[];
 
   /**
-   * The last entry has a NULL typeName.
+   * The last entry has a nullptr typeName.
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/vmStructs_jvmci.hpp:38]: null

File: src/hotspot/share/jvmci/vmStructs_jvmci.hpp [hunk 15]
```diff
@@ -30,24 +30,24 @@
...
    */
   static VMTypeEntry localHotSpotVMTypes[];
 
   /**
    * Table of integer constants.
-   * The last entry has a NULL typeName.
+   * The last entry has a nullptr typeName.
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/vmStructs_jvmci.hpp:44]: null

File: src/hotspot/share/jvmci/vmStructs_jvmci.hpp [hunk 16]
```diff
@@ -30,24 +30,24 @@
...
    */
   static VMIntConstantEntry localHotSpotVMIntConstants[];
 
   /**
    * Table of long constants.
-   * The last entry has a NULL typeName.
+   * The last entry has a nullptr typeName.
```
Comment (jdksjolen) [inline src/hotspot/share/jvmci/vmStructs_jvmci.hpp:50]: null

**Comment by jdksjolen** [comment]
Passes tier1.

File: src/hotspot/share/jvmci/jvmciCompilerToVM.cpp [hunk 17]
```diff
@@ -1237,11 +1237,11 @@ GrowableArray<Method*>* init_resolved_methods(jobjectArray methods, JVMCIEnv* JV
  */
 bool matches(jobjectArray methods, Method* method, GrowableArray<Method*>** resolved_methods_ref, Handle* matched_jvmci_method_ref, Thread* THREAD, JVMCIEnv* JVMCIENV) {
   GrowableArray<Method*>* resolved_methods = *resolved_methods_ref;
-  if (resolved_methods == NULL) {
+  if (resolved_methods == nullptr) {
     resolved_methods = init_resolved_methods(methods, JVMCIENV);
     *resolved_methods_ref = resolved_methods;
   }
-  assert(method != NULL, "method should not be NULL");
+  assert(method != nullptr, "method should not be nullptr");
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciCompilerToVM.cpp:1244]: `nullptr` in message.

File: src/hotspot/share/jvmci/jvmciEnv.cpp [hunk 18]
```diff
@@ -1472,14 +1472,14 @@ JVMCIObject JVMCIEnv::wrap(jobject object) {
 }
 
 jlong JVMCIEnv::make_oop_handle(const Handle& obj) {
-  assert(!obj.is_null(), "should only create handle for non-NULL oops");
+  assert(!obj.is_null(), "should only create handle for non-nullptr oops");
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciEnv.cpp:1475]: `nullptr` in message.

File: src/hotspot/share/jvmci/jvmciEnv.cpp [hunk 19]
```diff
@@ -1458,8 +1458,8 @@ Handle JVMCIEnv::asConstant(JVMCIObject constant, JVMCI_TRAPS) {
       JVMCI_THROW_MSG_(NullPointerException, "Foreign object reference has been cleared", Handle());
     }
     oop result = resolve_oop_handle(object_handle);
-    if (result == NULL) {
-      JVMCI_THROW_MSG_(InternalError, "Constant was unexpectedly NULL", Handle());
+    if (result == nullptr) {
+      JVMCI_THROW_MSG_(InternalError, "Constant was unexpectedly nullptr", Handle());
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciEnv.cpp:1462]: `nullptr` in message.

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp [hunk 20]
```diff
@@ -199,11 +199,11 @@ void HotSpotJVMCI::compute_offsets(TRAPS) {
     return resolve(object)->is_a(className::klass());                                   \
   }                                                                                     \
   void HotSpotJVMCI::className::check(oop obj, const char* field_name, int offset) {    \
-    assert(obj != NULL, "NULL field access of %s.%s", #className, field_name); \
+    assert(obj != nullptr, "nullptr field access of %s.%s", #className, field_name); \
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:202]: `nullptr` in message

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp [hunk 21]
```diff
@@ -225,28 +225,28 @@ void HotSpotJVMCI::compute_offsets(TRAPS) {
...
       oop base = ik->static_field_base_raw();                                                                             \
       oop result = HeapAccess<>::oop_load_at(base, className::_##name##_offset);                                          \
       return type(result);                                                                                                \
     }                                                                                                                     \
     void HotSpotJVMCI::className::set_##name(JVMCIEnv* env, type x) {                                                     \
-      assert(className::klass() != NULL && className::klass()->is_linked(), "Class not yet linked: " #className);         \
-      assert(className::klass() != NULL, "Class not yet loaded: " #className);                                            \
+      assert(className::klass() != nullptr && className::klass()->is_linked(), "Class not yet linked: " #className);         \
+      assert(className::klass() != nullptr, "Class not yet loaded: " #className);                                            \
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:236]: Align `\`

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp [hunk 22]
```diff
@@ -225,28 +225,28 @@ void HotSpotJVMCI::compute_offsets(TRAPS) {
...
       InstanceKlass* ik = className::klass();                                                                             \
       oop base = ik->static_field_base_raw();                                                                             \
       HeapAccess<>::oop_store_at(base, className::_##name##_offset, x);                                                   \
     }
 #define STATIC_PRIMITIVE_FIELD(className, name, jtypename)                                                                \
     jtypename HotSpotJVMCI::className::get_##name(JVMCIEnv* env) {                                                        \
-      assert(className::klass() != NULL && className::klass()->is_linked(), "Class not yet linked: " #className);         \
+      assert(className::klass() != nullptr && className::klass()->is_linked(), "Class not yet linked: " #className);         \
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:243]: Align `\`

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp [hunk 23]
```diff
@@ -225,28 +225,28 @@ void HotSpotJVMCI::compute_offsets(TRAPS) {
...
       InstanceKlass* ik = className::klass();                                                                             \
       oop base = ik->static_field_base_raw();                                                                             \
       return *base->field_addr<jtypename>(className::_##name##_offset);                                                   \
     }                                                                                                                     \
     void HotSpotJVMCI::className::set_##name(JVMCIEnv* env, jtypename x) {                                                \
-      assert(className::klass() != NULL && className::klass()->is_linked(), "Class not yet linked: " #className);         \
+      assert(className::klass() != nullptr && className::klass()->is_linked(), "Class not yet linked: " #className);         \
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:249]: Align `\`

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp [hunk 24]
```diff
@@ -266,7 +266,7 @@ void HotSpotJVMCI::compute_offsets(TRAPS) {
  *  oop HotSpotJVMCI::Architecture::wordKind(JVMCIEnv* env, oop obj) { ... }
  * void HotSpotJVMCI::Architecture::set_wordKind(JVMCIEnv* env, oop obj, oop x) { ... }
  *
- * InstanceKlass *HotSpotJVMCI::Architecture::_klass = NULL;
+ * InstanceKlass *HotSpotJVMCI::Architecture::_klass = nullptr;
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:269]: Right. This is exception case you listed: expression in comment.

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp [hunk 25]
```diff
@@ -621,11 +621,11 @@ JVMCI_CLASSES_DO(EMPTY2, EMPTY0, FIELD2, FIELD2, FIELD2, FIELD2, FIELD2, FIELD3,
     return jni()->IsInstanceOf(object.as_jobject(), className::clazz()) != 0;                                                     \
   }                                                                                                                               \
   void JNIJVMCI::className::check(JVMCIEnv* jvmciEnv, JVMCIObject obj, const char* field_name, jfieldID offset) {                 \
-    assert(obj.is_non_null(), "NULL field access of %s.%s", #className, field_name);                                     \
+    assert(obj.is_non_null(), "nullptr field access of %s.%s", #className, field_name);                                     \
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:624]: `nullptr` in message

File: src/hotspot/share/jvmci/jvmciJavaClasses.hpp [hunk 26]
```diff
@@ -391,8 +391,8 @@ class HotSpotJVMCI {
   static void check(JVMCIEnv* jvmciEnv, JVMCIObject obj, const char* field_name, jfieldID offset);  \
   static jclass _class;                                                                             \
 public:                                                                                             \
- static jclass clazz() { assert(_class != NULL, #fullClassName " uninitialized"); return _class; }  \
- static jclass fullClassName ##_class()  { assert(_class != NULL, "uninit"); return _class; }
+ static jclass clazz() { assert(_class != nullptr, #fullClassName " uninitialized"); return _class; }  \
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.hpp:394]: Should we align `\` in previous lines?

File: src/hotspot/share/jvmci/jvmciObject.hpp [hunk 27]
```diff
@@ -65,7 +65,7 @@ class JVMCIArray : public JVMCIObject {
 class JVMCIObjectArray : public JVMCIArray {
  public:
   JVMCIObjectArray() {}
-  JVMCIObjectArray(void* v): JVMCIArray() { assert(v == NULL, "must be NULL"); }
+  JVMCIObjectArray(void* v): JVMCIArray() { assert(v == nullptr, "must be nullptr"); }
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciObject.hpp:68]: `nullptr` in message

File: src/hotspot/share/jvmci/jvmciObject.hpp [hunk 28]
```diff
@@ -74,7 +74,7 @@ class JVMCIObjectArray : public JVMCIArray {
 class JVMCIPrimitiveArray : public JVMCIArray {
  public:
   JVMCIPrimitiveArray() {}
-  JVMCIPrimitiveArray(void* v): JVMCIArray() { assert(v == NULL, "must be NULL"); }
+  JVMCIPrimitiveArray(void* v): JVMCIArray() { assert(v == nullptr, "must be nullptr"); }
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciObject.hpp:77]: `nullptr` in message

**Review by vnkozlov — COMMENTED** [review]
Few missed `nullptr` in messages.

**Comment by jdksjolen** [comment]
Thanks @vnkozlov, I've fixed the issues you found!

**Comment by vnkozlov** [comment]
You need second review from JVMCI experts. @dougxc or @tkrodriguez please look.

**Review by tkrodriguez — APPROVED** [review]

**Comment by jdksjolen** [comment]
Thank you, integrating.

/integrate

**Pull request closed without merging**